src/nnet3bin/nnet3-get-egs.cc

```diff
@@ -76,15 +87,53 @@ static void ProcessFile(const MatrixBase<BaseFloat> &feats,
 
     // if applicable, add the iVector feature.
     if (ivector_feats != NULL) {
-      // try to get closest frame to middle of window to get
-      // a representative iVector.
-      int32 closest_frame = t + (actual_frames_per_eg / 2);
-      KALDI_ASSERT(ivector_feats->NumRows() > 0);
-      if (closest_frame >= ivector_feats->NumRows())
-        closest_frame = ivector_feats->NumRows() - 1;
-      Matrix<BaseFloat> ivector(1, ivector_feats->NumCols());
-      ivector.Row(0).CopyFromVec(ivector_feats->Row(closest_frame));
-      eg.io.push_back(NnetIo("ivector", 0, ivector));
+      if (ivector_interval == 0) {
+        // try to get closest frame to middle of window to get
+        // a representative iVector.
+        int32 closest_frame = t + (actual_frames_per_eg / 2);
+        KALDI_ASSERT(ivector_feats->NumRows() > 0);
+        if (closest_frame >= ivector_feats->NumRows())
+          closest_frame = ivector_feats->NumRows() - 1;
+        Matrix<BaseFloat> ivector(1, ivector_feats->NumCols());
+        ivector.Row(0).CopyFromVec(ivector_feats->Row(closest_frame));
+        eg.io.push_back(NnetIo("ivector", 0, ivector));
+      } else {
+        // try to get a representative iVector every ivector_interval frames
+        // in an eg.
+        KALDI_ASSERT(ivector_feats->NumRows() > 0);
+        // num_ivectors_left is the num of ivectors for frames whose t < 0.
+        // num_ivectors_right is the num of ivectors for frames whose t >= 0.
+        // They are computed according to how Round descriptor works.
```

Comment:
A brief description of round descriptor would help. At least suggest where one has to look for documentation about the Round descriptor.